commons/src/test/java/io/aiven/kafka/tieredstorage/commons/transform/FetchChunkEnumerationTest.java

```diff
@@ -0,0 +1,141 @@
+
+import org.apache.kafka.server.log.remote.storage.RemoteLogSegmentMetadata;
+
+import io.aiven.kafka.tieredstorage.commons.ChunkManager;
+import io.aiven.kafka.tieredstorage.commons.manifest.SegmentManifest;
+import io.aiven.kafka.tieredstorage.commons.manifest.SegmentManifestV1;
+import io.aiven.kafka.tieredstorage.commons.manifest.index.FixedSizeChunkIndex;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class FetchChunkEnumerationTest {
+    @Mock
+    ChunkManager chunkManager;
+    @Mock
+    RemoteLogSegmentMetadata remoteLogSegmentMetadata;
+
+    final FixedSizeChunkIndex chunkIndex = new FixedSizeChunkIndex(10, 100, 10, 100);
+    final SegmentManifest manifest = new SegmentManifestV1(chunkIndex, false, null);
+
+    // 10 @ chars
+    static final byte[] SAMPLE_CONTENT = "0123456789".getBytes();
+
+    // Test scenarios
+    // 1. Initialization
+    // 1.1. Invalid start position
+    @Test
+    void failsWhenLargerStartPosition() {
+        // Given
+        final SegmentManifest manifest = new SegmentManifestV1(chunkIndex, false, null);
+        // When
+        final int from = 1000;
+        final int to = from + 1;
+        // Then
+        assertThatThrownBy(
+            () -> new FetchChunkEnumeration(chunkManager, remoteLogSegmentMetadata, manifest, from, to))
+            .hasMessage("Invalid start position " + from + " in segment remoteLogSegmentMetadata");
+    }
+
+    // 1.2 End position within index
+    @Test
+    void endPositionIsWithinIndex() {
+        // Given
+        // a set of 10 chunks
+        // When
+        final int from = 0;
+        final int to = 80;
+        // Then
+        final FetchChunkEnumeration fetchChunk =
+            new FetchChunkEnumeration(chunkManager, remoteLogSegmentMetadata, manifest, from, to);
+        assertThat(fetchChunk.startChunkId).isEqualTo(0);
+        assertThat(fetchChunk.lastChunkId).isEqualTo(8);
+    }
+
+    // 1.3 End position outside index
+    @Test
+    void endPositionIsOutsideIndex() {
+        // Given
+        // a set of 10 chunks
+        // When
+        final int from = 0;
+        final int to = 110;
+        final FetchChunkEnumeration fetchChunk =
+            new FetchChunkEnumeration(chunkManager, remoteLogSegmentMetadata, manifest, from, to);
+        // Then
+        assertThat(fetchChunk.startChunkId).isEqualTo(0);
+        assertThat(fetchChunk.lastChunkId).isEqualTo(9);
+    }
+
+    // 2. Single chunk
+    @Test
+    void shouldReturnRangeFromSingleChunk() throws IOException {
+        // Given
+        // a set of 10 chunks
+        // When
+        final int from = 2;
+        final int to = 5;
```

Comment:
Ideally this range should not be in the first chunk since the range calculation is based on the chunk start and for the first chunk it is 0.

Reply:
Sure thing! Applying suggestions